src/plugins/owner/ECDSAValidationPlugin.sol

```diff
@@ -0,0 +1,112 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.25;
+
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {PackedUserOperation} from "@eth-infinitism/account-abstraction/interfaces/PackedUserOperation.sol";
+import {MessageHashUtils} from "@openzeppelin/contracts/utils/cryptography/MessageHashUtils.sol";
+
+import {IPlugin} from "../../interfaces/IPlugin.sol";
+import {IValidation} from "../../interfaces/IValidation.sol";
+import {BasePlugin} from "../BasePlugin.sol";
+import {PluginManifest, PluginMetadata} from "../../interfaces/IPlugin.sol";
+
+contract ECDSAValidationPlugin is IValidation, BasePlugin {
+    using ECDSA for bytes32;
+    using MessageHashUtils for bytes32;
+
+    uint256 internal constant _SIG_VALIDATION_PASSED = 0;
+    uint256 internal constant _SIG_VALIDATION_FAILED = 1;
+
+    // bytes4(keccak256("isValidSignature(bytes32,bytes)"))
+    bytes4 internal constant _1271_MAGIC_VALUE = 0x1626ba7e;
+    bytes4 internal constant _1271_INVALID = 0xffffffff;
+
+    mapping(uint8 id => mapping(address account => address)) public owners;
```

Comment:
Having an id only on the plugin won't expand the plugin validation function to be able to be installed on the account more than once without the account essentially stores validations differently (not use FunctionReference as a key). 

Plugins can build the storage whatever way they want as long as they dont violate the rules.

Like we discussed in the sync today, using an ID or not does not change if a validation plugin can be a layered validation (to be used by account's other validation functions).

Reply:
What I meant is that having the ID only on plugins won't enable the feature (multiple installation with the same validation plugin).
The account need to change how it organizes validation functions to enable this feature. 
In summary, this feature will require account changes and how account and validation plugin communicates (during installation). One example is [here](https://github.com/erc6900/reference-implementation/pull/87/files#diff-195373edfcb9ea525af4d8661a2446c68c317a6667c2ce38d5f242497faf58f3R87-R88).

Reply:
Oh I see, yes that will need to not be pulled from the manifest. The account-controlled auto assign mechanism works, or we can make it user-specified.